File: Spring Boot/demo/src/main/java/controller/GroupController.java
package controller;

import java.util.Hashtable;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import model.Group;
import model.Student;
import service.GroupService;
import service.StudentService;

@RestController
@RequestMapping("/groups")
public class GroupController {
	@Autowired
	StudentService ss;

	@Autowired
	GroupService gs;
	
	@RequestMapping("add")
	public Group addGroup(
			@RequestParam(name="id") String id,
			@RequestParam(name="code") String code,
			@RequestParam(name="entry") String entry
		) {
		
		gs.addGroup(id, code, entry);
		return gs.getGroup(id);
	}

	@RequestMapping("/all")
	public Hashtable<String, Group> getAll() {
		return gs.getAll();
	}
	
	@RequestMapping("{id}")
	public Group getGroup(@PathVariable("id") String id) {
		return gs.getGroup(id);
	}
	
	@RequestMapping("{id}/students")
	public Hashtable<String, Student> getStudents(@PathVariable("id") String id) {
		Hashtable<String, Student> students = new Hashtable<String, Student>();
		ss.getAll().forEach((k, v) -> {
			if (v.getGroup().equals(id)) {
				students.put(v.getId(), v);
			}
		});
		
		return students;
	}
}

File: LoopBack/src/controllers/group.controller.ts
import {
  repository,
  } from '@loopback/repository';
import {
  param,
  get,
} from '@loopback/rest';
import {Group, Student} from '../models';
import {GroupRepository, StudentRepository} from '../repositories';

export class GroupController {
  constructor(
    @repository(GroupRepository)
    public groupRepository : GroupRepository,
    @repository(StudentRepository)
    public studentRepository : StudentRepository,
  ) {}

  @get('/groups/add')
  async addGroup(
    @param.query.string('id') id: string,
    @param.query.string('code') code: string,
    @param.query.string('entry') entry: string,
  ): Promise<Group> {
    return await this.groupRepository.create({
      "id": id,
      "code": code,
      "entry": entry,
    });
  }

  @get('/groups/all')
  async getAll(): Promise<Group[]> {
    return await this.groupRepository.find();
  }

  @get('/groups/{id}')
  async getGroup(@param.path.string('id') id: string): Promise<Group> {
    return await this.groupRepository.findById(id);
  }

  @get('/groups/{id}/students')
  async getStudents(@param.path.string('id') id: string): Promise<Student[]> {
    const students = await this.studentRepository.find();
    return await students.filter(v => v.group === id);
  }
}
